uw-frame-components/portal/messages/partials/notifications-bell.html

```diff
@@ -18,73 +18,81 @@
-      <div class="notification-badge" aria-label="{{status}}" ng-class="{ 'notification-badge-empty' : vm.notifications.length === 0 }">
-        <span class="number-of-nots" ng-show="vm.notifications.length > 0"
+      <div class="notification-badge" aria-label="{{ vm.status }}" ng-class="{ 'notification-badge-empty' : vm.notifications.length === 0 }">
+        <span class="number-of-nots" ng-show="vm.notifications.length > 0 && !vm.isLoading"
               ng-class="{ 'more-than-10-nots' : (vm.notifications.length > 9) }">{{ vm.notifications.length }}</span>
         <md-icon>notifications</md-icon>
+        <md-progress-circular md-mode="indeterminate" ng-show="vm.isLoading" md-diameter="15px" class="md-primary"></md-progress-circular>
       </div>
     </md-button>
     <md-menu-content class="top-bar-menu-content notifications-menu">
       <md-menu-item layout="row" layout-align="space-between center">
         <span>Notifications</span>
         <a ng-href="{{ vm.notificationsUrl }}" ng-click="vm.pushGAEvent('Top Bar', 'Click Link', 'Notification Bell');" aria-label="see all notifications" class="link__see-all">
-          See all ({{ vm.notifications.length }})
+          See all <span ng-if="vm.notifications.length > 0">({{ vm.notifications.length }})</span>
         </a>
       </md-menu-item>
+      <!-- EMPTY STATE -->
+      <md-menu-item ng-show="vm.notifications.length === 0" class="top-bar-menu-item notifications-empty">
+        <div class="menu-item-content" layout="row" layout-align="center center">
+          All caught up!
+        </div>
+      </md-menu-item>
       <!-- SHOW UP TO 3 NEW NOTIFICATIONS -->
-      <md-menu-item ng-repeat="notification in vm.notifications | orderBy:'priority' | limitTo:3"
-                    class="top-bar-menu-item">
-        <!-- NOTIFICATION CONTENT -->
-        <div class="menu-item-content" layout="row" layout-align="space-between center">
-          <div flex="80" layout="column" layout-align="center start">
-            <a ng-href="{{ notification.actionButton.url ? notification.actionButton.url : vm.notificationsUrl }}">
-              <md-icon ng-if="notification.priority">priority_high</md-icon>
-              <span> {{ notification.title }}</span>
-            </a>
+      <div ng-show="vm.notifications.length > 0">
+        <md-menu-item ng-repeat="notification in vm.notifications | orderBy:'priority' | limitTo:3"
+                      class="top-bar-menu-item">
+          <!-- NOTIFICATION CONTENT -->
+          <div class="menu-item-content" layout="row" layout-align="space-between center">
+            <div flex="80" layout="column" layout-align="center start">
+              <!-- REGULAR MESSAGE -->
+              <span ng-if="!notification.priority">{{ notification.title }}</span>
+              <!-- PRIORITY MESSAGE -->
+              <div layout="row" layout-align="start center" ng-if="notification.priority">
+                <md-icon flex="10">priority_high</md-icon>
```

Comment:
This icon isn't just decorative; it's conveying meaning about the importance of the notification it adorns. Needs some kind of alt text?